src/components/AddEvent.tsx

```diff
@@ -88,90 +81,103 @@ const AddEvent = () => {
     setWhenToRecur(e.currentTarget.id);
   };
 
-  const formSubmitHandler = (e: React.FormEvent) => {
-    e.preventDefault();
-    const randomId = Math.floor((1 + Math.random()) * 0x10000)
+  const newEvent: Event = {
+    id: Math.floor((1 + Math.random()) * 0x10000)
       .toString(16)
-      .substring(1);
-
-    const randomColor = `#${Math.floor(Math.random() * 16777215).toString(16)}`;
-    const value = (endTime - startTime) / 60;
-    const newEvent: Event = {
-      id: randomId,
-      title: eventTitle,
-      value: value,
-      color: randomColor,
-      start: startTime,
-      end: endTime,
-      recurrence,
-    };
+      .substring(1),
+    title: eventTitle,
+    value: (endTime - startTime) / 60,
+    color: `#${Math.floor(Math.random() * 16777215).toString(16)}`,
+    start: startTime,
+    end: endTime,
+    recurrence,
+  };
 
-    const getDates = (startDate: string, stopDate: string) => {
-      const dateArray = [];
-      let currentDate = new Date(startDate);
-      while (currentDate <= new Date(stopDate)) {
-        switch (whenToRecur) {
-          case "day":
-            dateArray.push(fromDateToString(currentDate));
-            currentDate = currentDate.addDays(1);
-            break;
-          case "workday":
-            if (currentDate.getDay() > 0 && currentDate.getDay() < 6) {
-              dateArray.push(fromDateToString(currentDate));
-            }
-            currentDate = currentDate.addDays(1);
-            break;
-          case "weekend":
-            if (currentDate.getDay() === 0 || currentDate.getDay() === 6) {
-              dateArray.push(fromDateToString(currentDate));
-            }
-            currentDate = currentDate.addDays(1);
-            break;
-        }
+  //confirmation handler for overlapping events
+  const confirmReplaceHandler = () => {
+    const lastDate = new Date(selectedDate).addDays(parseInt(recurrence));
+    const dates = createDatesArray(selectedDate, lastDate, whenToRecur);
+    let newState = [...data];
```

Comment:
why let? you can use const when using array.push method. (find out more online)

Reply:
understood